shared/middleware/files_api.rb

```diff
@@ -487,10 +487,11 @@ def files_put_file(encrypted_channel_id, filename, body)
     end
     new_entry_hash = JSON.parse new_entry_json
     # Replace downcased filename with original filename (to preserve case in the manifest)
-    new_entry_hash['filename'] = filename
+    new_entry_hash['filename'] = CGI.unescape(filename)
```

Comment:
I don't think tests in shared contribute to coverage reports at the moment, only tests in dashboard. So the coverage we are seeing here is probably from what dashboard tests are generating.